scite/models.py

```diff
@@ -1,7 +1,19 @@
 # -*- coding: utf-8 -*-
-from typing import Any, AnyStr, Dict, List
+from typing import List
+from pydantic import BaseModel, constr, validator
 
 
-DOI = AnyStr
-DOIs = List[DOI]
-JSON = Dict[str, Any]
\ No newline at end of file
+class DOI(BaseModel):
+    id: constr(regex=r"^10\.\d{4,9}/[-._;()/:a-zA-Z0-9]+$")

```

Comment:
This is something I found on the web - Crossref [uses](https://www.crossref.org/blog/dois-and-matching-regular-expressions/) this pattern so I think it should be good enough.